client/db/src/lib.rs

```diff
@@ -1316,26 +1291,18 @@ impl<Block: BlockT> Backend<Block> {
 
 			transaction.set_from_vec(columns::HEADER, &lookup_key, pending_block.header.encode());
 			if let Some(body) = pending_block.body {
-				match self.transaction_storage {
-					TransactionStorageMode::BlockBody => {
-						transaction.set_from_vec(columns::BODY, &lookup_key, body.encode());
-					},
-					TransactionStorageMode::StorageChain => {
-						let body =
-							apply_index_ops::<Block>(&mut transaction, body, operation.index_ops);
-						transaction.set_from_vec(columns::BODY, &lookup_key, body);
-					},
+				// If we have any index operations we save block in the new format with indexed
+				// extrinsic headers Otherwise we save the body as a single blob.
```

Comment:
Could be an idea to always store as`DBExtrinsic` in BODY_INDEX so reading in BODY is just for compatibility,  and a later migration could be done.
But this can be switched any time, and this save some disk space.

Reply:
Decided against that for now. If at some point We'll decide that each extrinsic needs to be indexed we can make the new format the default.